src/extension.ts

```diff
@@ -37,7 +47,15 @@ export function activate(context: vscode.ExtensionContext) {
 
             const text = editor.document.getText()
             // JSON.parse(text)
-            const path = jsPathTo(text, editor.document.offsetAt(editor.selection.active), nonQuotedKeyRegex)
+            let path = jsPathTo(text, editor.document.offsetAt(editor.selection.active), nonQuotedKeyRegex)
+
+            if (fileNameAsPrefix && editor.document.fileName) {
+                const fileName = parse(editor.document.fileName).base;
+                const fileNameWithoutExtension = fileName.substring(0, fileName.lastIndexOf('.'))
```

Comment:
What if the file doesn't have an extension? For example prettier can be configured with JSON `.prettierrc` file (https://prettier.io/docs/en/configuration.html). In this case file name is empty. I expect there can be similar JSON files not starting with a dot as well. Could you take these cases into account?

Reply:
Fixed